### PR TITLE
enhance contribution guidelines

### LINKED ISSUE #214: enhance contribution guidelines
The current approach described in the CONTRIBUTING.md and the usage of waffle.io is quite awesome and provides an easy and powerful workflow. There's just one case where it doesn't fit its needs:

Imagine a story describing one feature. Any ticket should have an assignee and all resulting commits **must** be squashed before merging in order to avoid a big history with much commits like "fixed type (again)...".

When having stories this causes too large diffs with a non-reliable blame feature.

Therefore I'd like to propose a change in the contribution guidelines for stories:

In case of a story the ticket **MUST** not have an assignee. The "real" stories are just comming and it is quite impossible to have just one working on it.
In case of a story the commits should **NOT** be squashed. As a story contains a lot of sub-tasks written in the PR of the story. (when having ten "minor review fixes" commits, they should be squashed, but not sub-tasks").

When having the following tasks: 'import deps for $feature', 'build api and socket connector for $feature', 'implement frontend based on reactjs for $feature', 'use server-side rendering for $feature' then the story should have a commit for each of those subtasks squashed.

The repo owner (in this case "me") should squash and merge story PRs after a big review properly to fulfill the provided rules.

Every story **MUST** have at least two reviewers (most of the tickets are not that heavy, so one reviewer is enough, but stories introduce great changes, so two reviewers are necessary)

### PR DESCRIPTION
**Pull request by benbieler**
Resolves #214

### CODE REVIEW

**Comment by benbieler** [comment]
@Ma27, CI can be very helpfull, and prevent massive mistakes from happening, but in cases like these, where soly a markdown file has been changed, is there no way to turn it off?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -36,6 +36,14 @@ So if you add a realtime messenger to this application, the branch could look li
 
     short-description
 
+### Stories
+
+####When assigning yourself to a ticket
+
+- Stories **MUST** not have an assignee. This is due to the fact, that it is quite impossible to work on big ones by yourself.
+- In case of a story the commits should **NOT** be squashed, as a story contains a lot of sub-tasks(which on the other hand should be squashed) written in the PR of the story.
```
Comment (Ma27) [inline CONTRIBUTING.md]: before the opening brace a space is missing

**Comment by Ma27** [comment]
I've created an issue for that, see #227 
As noone is assigned to this, anyone can look at this if he would find a suitable solution for your (definetly valid) issue.

You should know that I've changed the settings so only PRs can change `master` and any CI should be happy with those PRs. That's kind of a complex solution, but it prevents me from broken code on master (but well, the tests are quite fishy - there are still issues with false-positives, but there are issues for it)

**Comment by Ma27** [comment]
squash please

**Comment by Ma27** [comment]
@benbieler I'm currently working on #228 
IN this PR I've just refactored the eslint config (eslint 2.x has been released two days ago and some breaking changes are causing the red CI). WHen this is merged, please rebase your PRs and then feel free to merge if everything gets green :)

**Comment by Ma27** [comment]
btw I've just cleared all assignees from all current stories being in the backlog, the tasks should be defined for each story when starting the progress

**Comment by Ma27** [comment]
@benbieler I'll finish this one as I require this for #231

**Pull request merged**